src/core/ddsi/src/q_entity.c

```diff
@@ -1807,7 +1807,7 @@ static void reader_add_connection (struct reader *rd, struct proxy_writer *pwr,
       /* FIXME: for now, assume that the ports match for datasock_mc --
        't would be better to dynamically create and destroy sockets on
        an as needed basis. */
-      ddsi_join_mc (rd->e.gv, rd->e.gv->mship, rd->e.gv->data_conn_mc, &m->ssm_src_loc, &m->ssm_mc_loc);
+      (void) ddsi_join_mc (rd->e.gv, rd->e.gv->mship, rd->e.gv->data_conn_mc, &m->ssm_src_loc, &m->ssm_mc_loc);
```

Comment:
No, casting away this return value is a mistake. An error here is definitely possible and should be handled properly. It probably means the entire call needs to fail, that might mean the proxy writer needs to be undiscovered or something like that, but it being difficult to figure out _how_ to handle an error is not a sufficient reason to pretend it can't occur.

Reply:
I will check the return code and log a message.